plugins/inputs/consul_metrics/consul_metrics.go

```diff
@@ -0,0 +1,200 @@
+package consul_metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Consul_metrics configuration object
+type ConsulMetrics struct {
+	URL string `toml:"url"`
+
+	TokenFile string `toml:"token_file"`
+	Token     string `toml:"token"`
+
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+
+	tls.ClientConfig
+
+	roundTripper http.RoundTripper
+}
+
+const timeLayout = "2006-01-02 15:04:05 -0700 MST"
+
+var sampleConfig = `
+  ## URL for the Consul agent
+  # url = "http://127.0.0.1:8500"
+
+  ## Use auth token for authorization. 
+  ## Only one of the options can be set. Leave empty to not use any token.
+  # token_file = "/path/to/auth/token"
+  ## OR
+  # token = "a1234567-40c7-9048-7bae-378687048181"
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = /path/to/cafile
+  # tls_cert = /path/to/certfile
+  # tls_key = /path/to/keyfile
+`
+
+func init() {
+	inputs.Add("consul_metrics", func() telegraf.Input {
+		return &ConsulMetrics{
+			ResponseTimeout: config.Duration(5 * time.Second),
+		}
+	})
+}
+
+// SampleConfig returns a sample config
+func (n *ConsulMetrics) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a description of the plugin
+func (n *ConsulMetrics) Description() string {
+	return "Read metrics from the Consul API"
+}
+
+func (n *ConsulMetrics) Init() error {
+	if n.URL == "" {
+		n.URL = "http://127.0.0.1:8500"
+	}
+
+	if n.TokenFile != "" && n.Token != "" {
+		return fmt.Errorf("config error: both token_file and token are set")
```

Comment:
Please use `errors.New` if you do not pass in a formatting string and variable...
```suggestion
		return errors.New("config error: both token_file and token are set")
```

Reply:
You're right here

Reply:
Soooo? :-) Can you please change the code accordingly?